Fixes #257 by refactoring import autocomplete/conversion code.

coreystaten opened the pull request:
This adds import conversion on paste, as well as conversion on changing focus away from editor.

Some remaining deficiencies:
* When pasting imports, the cursor ends up after the last import converted instead of at the end of the pasted text.
* Clicking elsewhere in the same editor doesn't convert imports.

zjmiller commented:
I think one refactor that makes sense here is to move as much as possible into a separate file, or separate files.

zjmiller commented:
One more thing, it would be nice if this worked with submitting new subquestions as well. For example, if a user types "Analyze $1" in the new subquestion form and then clicks Submit. This currently doesn't appear to be working in this PR. I checked this branch out locally and was able to get it to work by wrapping the handler associated with submitting a subquestion in `setTimeout`. Maybe there's a more elegant way to do this.

coreystaten commented:
Fixed the bug you pointed out, refactored the code into a separate file, and added a global import conversion when the editor component mounts.

Re: submitting a new question, the current won't fix wouldn't work if the newly displayed editors were read only (since those wouldn't use the BlockEditorEditing component) --  a solution for that would be slightly more work.
